runtime/lua/vim/fs.lua

```diff
@@ -334,6 +327,46 @@ function M.find(names, opts)
   return matches
 end
 
+--- Find the first parent directory containing a specific "marker", relative to a buffer's
+--- directory.
+---
+--- Example:
+---
+--- ```lua
+--- -- Find the root of a Python project
+--- vim.fs.findroot(0, {'pyproject.toml', 'setup.py' })
+---
+--- -- Find the root of a git repository
+--- vim.fs.findroot(0, '.git')
+---
+--- -- Find the parent directory containing any file with a .csproj extension
+--- vim.fs.findroot(0, function(name, path)
+---   return name:match('%.csproj$') ~= nil
+--- end)
+--- ```
+---
+--- @param bufnr integer Buffer number, or 0 for current buffer.
+--- @param marker (string|string[]|fun(name: string, path: string): boolean) A marker, or list
+---               of markers, to search for. If a function, the function is called for each
+---               evaluated item and should return true if {name} and {path} are a match.
+--- @return string? # Directory containing one of the given markers, or nil if no directory was
+--- found.
+function M.findroot(bufnr, marker)
+  assert(bufnr, 'missing required argument: bufnr')
+  assert(marker, 'missing required argument: marker')
+  local paths = M.find(marker, {
+    upward = true,
+    path = vim.fs.dirname(vim.api.nvim_buf_get_name(bufnr)),
+    stop = vim.uv.os_homedir(),
```

Comment:
I'd tend to use the cwd as ceiling. No strong opinion on it, but maybe an indicator that this should be an optional parameter as well?

Reply:
I don't think `$HOME` is also good to stop at. At work most of our files go into NFS mounts (including XDG_CONFIG). We should just stop at `/` IMO.

Reply:
Removed `stop`, we'll just go to the root directory. 8dca2167f5f6fe578a15211aa6cf4fb76b1c8d32